File: labellab-server/models/detection.js
const mongoose = require('mongoose')

const DetectionSchema = new mongoose.Schema({
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User'
    },
    imageUrl: {
        type: String,
        required: true
    },
    detections: [{
        confidence: Number,
        label_name: ObjectId,
        x1: Number,
        y1: Number,
        x2: Number,
        y2: Number
    }],
    createdAt: {
        type: Date,
        default: Date.now
    }
})

module.exports = mongoose.model('Detection', DetectionSchema)

File: labellab_mobile/lib/model/detection.dart

class Detection {
  String id;
  String name;
  int confidence;
  int x1;
  int y1;
  int x2;
  int y2;

  Detection();

  Detection.fromJson(dynamic json) {
    id = json["_id"];
    name = json["label_name"];
    confidence = json["confidence"];
    x1 = json["x1"];
    y1 = json["y1"];
    x2 = json["x2"];
    y2 = json["y2"];
  }
}
